### ISSUE TITLE
Review `new_session` benchmark

### ISSUE THREAD

**Issue by apopiak** [paritytech/cumulus#644]
https://github.com/paritytech/cumulus/blob/caa8b8172b39c46c916cb4d0036159d05fcd65e3/pallets/collator-selection/src/benchmarking.rs#L154-L156

@JesseAbram I assume you wrote this? Could you expand on it?

**Comment by JesseAbram** [comment]
Yes most likely, I think the TODO can be removed. However, I am happy to explain the benchmark.

There are two O(n) operations based on how many candidates there are [here](https://github.com/paritytech/cumulus/blob/caa8b8172b39c46c916cb4d0036159d05fcd65e3/pallets/collator-selection/src/lib.rs#L405)

and based on how many candidates are removed [https://github.com/paritytech/cumulus/blob/caa8b8172b39c46c916cb4d0036159d05fcd65e3/pallets/collator-selection/src/lib.rs#L415](url)
